app/models/article.rb

```diff
@@ -565,6 +566,12 @@ def past_or_present_date
     end
   end
 
+  def canonical_url_must_not_have_spaces
+    if canonical_url.to_s.match(/[[:space:]]/).present?
```

Comment:
Note that there's also a `match?` predicate method:

```ruby
canonical_url.to_s.match?(/[[:space:]]/)
```

Reply:
I ran the test and they passed but I can change it, so should I?

Reply:
Up to you, but I'd prefer the `match?` form and since this PR hasn't been approved yet updating it won't invalidate any reviews. So if you like this suggestion, go ahead and change it 😃

Reply:
it looks better! thanks @citizen428